crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -85,51 +121,127 @@ pub(crate) struct TransactionBlockEffects {
     pub gas_effects: GasEffects,
     pub status: ExecutionStatus,
     pub errors: Option<String>,
-    // pub transaction_block: TransactionBlock,
-    // pub dependencies: Vec<TransactionBlock>,
-    // pub lamport_version: Option<u64>,
+
+    #[graphql(skip)]
+    pub tx_block_digest: Digest,
+    // pub transaction_block: Option<Box<TransactionBlock>>,
+    #[graphql(skip)]
+    pub dependencies: Vec<TransactionDigest>,
+    pub lamport_version: Option<u64>,
+    // unclear what object reads is about, TODO @ashok
     // pub object_reads: Vec<Object>,
     // pub object_changes: Vec<ObjectChange>,
-    // pub balance_changes: Vec<BalanceChange>,
-    // pub epoch: Epoch
-    // pub checkpoint: Checkpoint
+    pub balance_changes: Option<Vec<Option<BalanceChange>>>,
+    // have their own resolvers in the impl block
+    #[graphql(skip)]
+    pub epoch_id: u64,
+    // pub epoch: Option<Epoch>,
+    // pub checkpoint: Option<Checkpoint>,
 }
 
-impl From<&SuiTransactionBlockEffects> for TransactionBlockEffects {
-    fn from(tx_effects: &SuiTransactionBlockEffects) -> Self {
+impl TransactionBlockEffects {
+    pub fn from_stored_transaction(
+        balance_changes: Vec<Option<Vec<u8>>>,
+        _object_changes: Vec<Option<Vec<u8>>>,
+        tx_effects: &SuiTransactionBlockEffects,
+        tx_block_digest: Digest,
+    ) -> Option<Self> {
         let (status, errors) = match tx_effects.status() {
             SuiExecutionStatus::Success => (ExecutionStatus::Success, None),
             SuiExecutionStatus::Failure { error } => {
                 (ExecutionStatus::Failure, Some(error.clone()))
             }
         };
+        let lamport_version = tx_effects
```

Comment:
Not sure if this the right implementation for lamport version. Can we please double check?

Reply:
im inclined to think this is correct due to https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/execution.rs#L134 but will also defer